### ISSUE TITLE
Decoding nested objects from JSON fails

### ISSUE THREAD

**Issue by jbardin** [hashicorp/hcl#153]
A single JSON object with a single key and value of a another object fails to decode.

```
{
  "vars": {
    "var_1": {
      "default": {
        "key1": "a",
        "key2": "b"
      }
    }
  }
}
```

can't decode into a list of 
```
	type V struct {
		Name        string `hcl:",key"`
		Description string
		Default     map[string]string
	}
```

Adding a value for the Description field will change the structure enough to work

**Comment by jbardin** [comment]
For reference, parsing the above yields as ast that looks like: https://gist.github.com/jbardin/b1c38aa9c7080c520e8b9fb672bebe42#file-flattened

Whereas in this case we expect an ast that looks like:
https://gist.github.com/jbardin/b1c38aa9c7080c520e8b9fb672bebe42#file-expanded

**Comment by jbardin** [comment]
In this example, the "another" field is skipped entirely by the parser

```
{
    "var_1": {
      "default": {
        "B1": "a",
        "B2": "b"
      }
      "another": "field",
    }
  }
```

**Comment by jbardin** [comment]
fixed via #152
